backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/ReleaseService.java

```diff
@@ -16,21 +18,40 @@
 @Service
 public class ReleaseService implements ReleaseAmendingLawAndAllRelatedTargetLawsUseCase {
   private final LoadAmendingLawPort loadAmendingLawPort;
+  private final LoadAnnouncementPort loadAnnouncementPort;
   private final UpdateAmendingLawPort updateAmendingLawPort;
+  private final UpdateAnnouncementPort updateAnnouncementPort;
 
   public ReleaseService(
-      LoadAmendingLawPort loadAmendingLawPort, UpdateAmendingLawPort updateAmendingLawPort) {
+      LoadAmendingLawPort loadAmendingLawPort,
+      LoadAnnouncementPort loadAnnouncementPort,
+      UpdateAmendingLawPort updateAmendingLawPort,
+      UpdateAnnouncementPort updateAnnouncementPort) {
     this.loadAmendingLawPort = loadAmendingLawPort;
+    this.loadAnnouncementPort = loadAnnouncementPort;
     this.updateAmendingLawPort = updateAmendingLawPort;
+    this.updateAnnouncementPort = updateAnnouncementPort;
   }
 
   @Override
   public Optional<AmendingLaw> releaseAmendingLaw(Query query) {
-    final Optional<AmendingLaw> amendingLawOptional =
+    var amendingLawOptional =
         loadAmendingLawPort.loadAmendingLawByEli(new LoadAmendingLawPort.Command(query.eli()));
+
+    var announcementOptional =
+        loadAnnouncementPort.loadAnnouncement(new LoadAnnouncementPort.Command(query.eli()));
+    announcementOptional.flatMap(
+        announcement -> {
+          announcement.setReleasedByDocumentalistAt(Instant.now());
+
+          return updateAnnouncementPort.updateAnnouncement(
+              new UpdateAnnouncementPort.Command(announcement));
+        });
+
     return amendingLawOptional.flatMap(
```

Comment:
Does this now always return the amendingLaw, even in the "annoucmenets" case?

Reply:
it is just ensuring the announcement is kept in sync with the existing amending law. So yes it will always return the amending law. That the announcement is also updated should be transparent to the uses.
The release service will later be changed for the changes to the domain model.